### ISSUE TITLE
Health isn't displayed when specing as spectator

### ISSUE THREAD

**Issue by ETLegacyTeam** [etlegacy/etlegacy#941]
Author: @rmarquis (Spyhawk)
Date: 2016-09-15
Redmine Issue: 941, https://dev.etlegacy.com/issues/941

---

Player health is "empty" when specing as TEAM_SPECTATOR.
The issue probably also apply to constructible health (tank, truck, etc.)

**Comment by ETLegacyTeam** [comment]
From: @thunderpwn (thunder)
Date: 2016-09-16T17:45:59Z

---

shouldnt this be as it is.. thinking a bit about cheating here.. I know its kind of stupid but if someone is speccing a guy and helping out a friend that is playing he can tell him how much health the enemy has left?

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2016-09-16T17:55:39Z

---

If cheating is an issue, then spec shouldn't be allowed at all. Position is a much better info that health value :D

No, cheating is prevented with the speclock feature or private game. This is a bug.

**Comment by ETLegacyTeam** [comment]
From: @IR4T4 (IR4T4)
Date: 2016-12-01T14:52:58Z

---

I can't reproduce.

**Comment by ETLegacyTeam** [comment]
From: @saukko28 (Saukko)
Date: 2019-05-10T11:40:09Z

---

Alright, I investigated this a little and I found out that this happens only when you spectate a bot. When you spectate a real client who hovers over bot, it will show the healthbar correctly.

How to reproduce:

1. Start two clients
2. Start localhost with one of the clients and join to the created localhost with the other
3. Put another client to Allies or Axis and leave the other one into spectators
4. Spectate the client who you just put into either playing team

After these steps, when you hover the aim over a bot with the playing client and look with the other, you will noticed a full healthbar.

-*S
